messaging/controller/sender/sender.go

```diff
@@ -62,3 +73,34 @@ func (s *Sender) processAndMarshalMessageSpec(spec *encryption.ProtocolMessageSp
 
 	return messageBytes, sharedSecretKey, nil
 }
+
+func (s *Sender) telemetryTraceMessage(ctx context.Context, hashes [][]byte) context.Context {
+	if !s.tracer.Enabled() {
+		return ctx
+	}
+
+	funcName := "Sender.unknown"
+	pc, _, _, ok := runtime.Caller(1)
+	if ok {
+		fn := runtime.FuncForPC(pc)
+		if fn != nil {
+			funcName = fn.Name()
+		}
+	}
```

Comment:
Looks like this can be extracted into some utility function
```go
func callerFuncName(defaultValue string) string {
	pc, _, _, ok := runtime.Caller(2)
	if !ok {
        return defaultValue
    }
	fn := runtime.FuncForPC(pc)
	if fn == nil {
        return defaultValue
	}
    return fn.Name()
}